### PR TITLE
Use MobX to manage application state

### LINKED ISSUE #62: use mobx for state management
Mobx (https://mobx.js.org/) is a simple and powerful state management tool, especially when using React.

### PR DESCRIPTION
**Pull request by sgratzl**
closes #62 

extracted the state to an own class and let it be managed with mobx.

### CODE REVIEW

File: src/components/graph-vis/index.js [hunk 1]
```diff
@@ -1,207 +1,150 @@
...
-    }
+    // dummy render function in GeoJSSceneManager
+    // const animate = () => {
+    //   if (!this.scene.parent || !this.scene.dp) {
+    //     return;
+    //   }
+    //   this.scene.render();
+    //   window.requestAnimationFrame(animate);
+    // }
+    // window.requestAnimationFrame(animate);
```
Comment (waxlamp) [inline src/components/graph-vis/index.js]: If this code block is not needed anymore, could you just remove it?

File: src/geojs-scene-manager/index.js [hunk 2]
```diff
@@ -118,7 +123,7 @@ export class GeoJSSceneManager {
       onNode = true;
 
       const name = evt.data;
-      const node = nodes[name];
+      // const node = nodes[name];
```
Comment (waxlamp) [inline src/geojs-scene-manager/index.js]: Remove.

**Comment by waxlamp** [comment]
> This looks very cool.
> 
> In addition to the minor changes requested in my review, could you also replace the README file with an example of how to, say, add a value to the mobx store in development?

Also, the show/hide button for links does not seem to be working--could you see what's going on there?

**Comment by sgratzl** [comment]
> Also, the show/hide button for links does not seem to be working--could you see what's going on there?

fixed. was using an non-existent property. btw. this could have been prevented using TypeScript.

**Comment by sgratzl** [comment]
> In addition to the minor changes requested in my review, could you also replace the README file with an example of how to, say, add a value to the mobx store in development?

see https://github.com/arclamp/materialnet/blob/5124e13fb7291c238457dfd8e49b434e211acad7/README.md

**Pull request merged**